lib/srv/discovery/discovery.go

```diff
@@ -45,16 +47,32 @@ import (
 
 var errNoInstances = errors.New("all fetched nodes already enrolled")
 
+type Clients struct {
+	Cloud      cloud.Clients
+	Kubernetes kubernetes.Interface
+}
```

Comment:
I think they are better bundled, as they are both clients. Similar to how we bundled matchers. There might be some other clients in the future so it makes sense to have it there. I agree that maybe it was better to have separate PR for this bundling, but since it's already done here 🤷

Reply:
I agree with Tim. Sounds better to have them separate. I suggested this as a second option but I have preference about the first: separate cloud and kube clients

Reply:
I've unbundled them 👍